x-pack/plugin/esql/qa/testFixtures/src/main/resources/stats.csv-spec

```diff
@@ -1188,6 +1188,72 @@ e:i | l:i
 4   | 3
 ;
 
+nestedAggsOverGroupingWithAlias#[skip:-8.13.99,reason:supported in 8.14]
+FROM employees
+| STATS e = length(f) + 1, count(*) by f = first_name
```

Comment:
We should not allow the group key inside stats since its redundant, we already attach the key to the output.
Thanks for the queries - it looks like the Verifier could be improved.
I tried to keep things close to Postgres and, in case of expressions inside grouping, did not semantic matching rather basic equality.

`FROM employees | STATS  languages + emp_no BY languages + emp_no` works because the grouping key is easily found however we shouldn't allow this type of query.
`FROM employees | STATS e = languages + emp_no BY languages + emp_no` does not work because languages + emp_no is aliased and not matched with the grouping - better to do STATS e BY e = languages + emp_no.
However as I said, I'd prefer for us to prevent this and ideally have a better message ("the group key is already included in the output").

Reply:
Added more tests and updated the behavior.